Library/ViewModels/RewardAddOnSelectionViewModel.swift

```diff
@@ -256,15 +256,55 @@ public final class RewardAddOnSelectionViewModel: RewardAddOnSelectionViewModelT
     .map(PledgeViewData.init)
     .takeWhen(self.continueButtonTappedProperty.signal)
 
+    let defaultShippingTotal = Signal.zip(project, baseReward)
```

Comment:
The refactor would result to duplicate we are trying to avoid, because `PledgeViewModel` uses the result of it's calculation else where in the code.

Reply:
Not sure I understand what you mean. We don't need to replicate the logic of the PledgeViewModel everywhere. My understanding is that the shipping, at this point, should only be the shipping cost associated with the base reward since the add on has not been selected yet. But again, clarify with Laurie about whether this event should handle only new pledges or both new pledges + edits.